pkg/ansible/watches/watches.go

```diff
@@ -121,9 +135,55 @@ func (w *Watch) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
-// Load - loads a slice of Watches from the watch file at `path`.
-func Load(path string) ([]Watch, error) {
-	b, err := ioutil.ReadFile(path)
+// Validate - ensures that a Watch is valid
+// A Watch is considered valid if it:
+// - Specifies a valid path to a Role||Playbook
+// - If a Finalizer is non-nil, it must have a name + valid path to a Role||Playbook or Vars
+func (w *Watch) Validate() error {
+	err := verifyAnsiblePath(w.Playbook, w.Role)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Invalid ansible path for GVK: %v", w.GroupVersionKind.String()))
+		return err
+	}
+
+	if w.Finalizer != nil {
+		if w.Finalizer.Name == "" {
+			err = fmt.Errorf("finalizer must have name")
+			log.Error(err, fmt.Sprintf("Invalid finalizer for GVK: %v", w.GroupVersionKind.String()))
+			return err
+		}
+		// only fail if Vars not set
+		err = verifyAnsiblePath(w.Finalizer.Playbook, w.Finalizer.Role)
+		if err != nil && len(w.Finalizer.Vars) == 0 {
+			log.Error(err, fmt.Sprintf("Invalid ansible path on Finalizer for GVK: %v", w.GroupVersionKind.String()))
+			return err
+		}
+	}
+
+	return nil
+}
+
+// New - returns a Watch with sensible defaults.
+// The returned Watch is not valid as it does not include a role or playbook.
+func New(gvk schema.GroupVersionKind) *Watch {
+	reconcilePeriod, _ := time.ParseDuration(reconcilePeriodDefault)
+	return &Watch{
+		GroupVersionKind:            gvk,
+		MaxRunnerArtifacts:          maxRunnerArtifactsDefault,
+		MaxWorkers:                  maxWorkersDefault,
+		ReconcilePeriod:             reconcilePeriod,
+		ManageStatus:                manageStatusDefault,
+		WatchDependentResources:     watchDependentResourcesDefault,
+		WatchClusterScopedResources: watchClusterScopedResourcesDefault,
+	}
+}
+
+// Load - loads a slice of Watches from the watches file from the CLI
+func Load(flags *ansibleflags.AnsibleOperatorFlags) ([]Watch, error) {
+	// set the defaults used when unmarshalling
+	maxWorkersDefault = flags.MaxWorkers
```

Comment:
Again this smells to me that we are passing this thing around on a watch when I don't think it should be on a watch.